datasource/vmware/vmware_test.go

```diff
@@ -165,10 +182,36 @@ func TestFetchUserdata(t *testing.T) {
 			},
 			err: errors.New(`Unsupported encoding "test encoding"`),
 		},
+		{
+			variables: map[string]string{
+				"coreos.config.url": ts.URL + "/good",
+			},
+			userdata: "test config",
+		},
+		{
+			variables: map[string]string{
+				"coreos.config.url": ts.URL + "/bad",
+			},
+			err: pkg.ErrNotFound{fmt.Errorf("Not found. HTTP status code: %d", http.StatusNotFound)},
+		},
+		{
+			variables: map[string]string{
+				"coreos.config.url": ts.URL + "/ugly",
+			},
+			err: pkg.ErrServer{fmt.Errorf("Server error. HTTP status code: %d", http.StatusInternalServerError)},
+		},
+	}
+
+	var downloader urlDownloadFunction = func(url string) ([]byte, error) {
```

Comment:
It would be simpler if this just did a map lookup instead of actually creating an HTTP server and client. As it exists, the errors in this test would have to be kept in sync with the http package. While unlikely that those would change, the coupling is still unnecessary.